### PR TITLE
Implementation of reg v2 for shipyard

### PR DESCRIPTION
**Pull request by borzamircea**
- Ability to add registries with and without credentials
- Inspect a registry to see its images 
- Inspect an image to see its layers

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ image: media build
 release: build image
 	@docker push shipyard/shipyard:$(TAG)
 
-test: clean 
+test: clean
```
Comment (soninob) [inline Makefile]: please don't push removal or adding of spaces

File: auth/auth.go [hunk 2]
```diff
@@ -2,10 +2,9 @@ package auth
 
 import (
 	"errors"
+	"golang.org/x/crypto/bcrypt"
```
Comment (soninob) [inline auth/auth.go]: Is there a meaning for the order of import? If not, let's not push unnecessary changes

File: auth/builtin/builtin.go [hunk 3]
```diff
@@ -1,8 +1,8 @@
 package builtin
 
 import (
-	"golang.org/x/crypto/bcrypt"
 	"github.com/shipyard/shipyard/auth"
+	"golang.org/x/crypto/bcrypt"
```
Comment (soninob) [inline auth/builtin/builtin.go]: Is there a meaning for the order of import? If not, let's not push unnecessary changes

File: controller/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,5 @@
 FROM alpine:latest
+
```
Comment (soninob) [inline controller/Dockerfile]: Please don't push removal or adding of spaces

File: Makefile (hunk 1 above)
Comment (borzamircea) [inline Makefile]: it was go formatting

File: auth/auth.go (hunk 2 above)
Comment (borzamircea) [inline auth/auth.go]: go formatting again, in alphabetical order

File: auth/builtin/builtin.go (hunk 3 above)
Comment (borzamircea) [inline auth/builtin/builtin.go]: go formatting again, in alphabetical order

File: controller/Dockerfile (hunk 4 above)
Comment (borzamircea) [inline controller/Dockerfile]: ok
